### PR TITLE
Update payment meta methods, storage

### LINKED ISSUE #11: 3.0 - Migrate payment meta to order meta, use new order meta functions
The extension meta should be migrated to its own meta key and use order meta functions for EDD 3.0. There is also at least one use of `update_post_meta` on a payment, which will not work in 3.0.

The extension is also using the `edd_payment_meta` filter to add its metadata to the `_edd_payment_meta` array, and if I remember correctly, that doesn't work in 3.0--I feel like I had to change that in Simple Shipping.

### PR DESCRIPTION
**Pull request by robincornett**
Fixes #11 

Proposed Changes:
1. Add an EDD 3.0 order meta function to store campaign information.
2. Update the campaign info rendering method to retrieve 3.0 order metadata and convert old metadata if it exists.
3. Add a migration function for sites updating to EDD 3.0 to automatically move payment meta to new order meta. (pending)

### CODE REVIEW

File: includes/campaign-logger.php [hunk 1]
```diff
@@ -20,7 +20,43 @@ class EDDCT_Campaign_Logger {
...
+		}
+
+		return $payment_meta;
+	}
+
+	/**
+	 * Adds the campaign info to the order meta in EDD 3.0.
+	 *
+	 * @since 1.0.2
+	 * @param int $order_id
```
Comment (LisaCee) [inline includes/campaign-logger.php:41]: Will you add a description for this param?

**Review by LisaCee — CHANGES_REQUESTED** [review]
Just the one comment for now. 

I'm guessing to test, I migrate a site with Campaign Tracker from 2.x > 3 and double check that the meta is correct...

**Comment by robincornett** [comment]
@LisaCee yes, if you migrate orders on a site with campaign tracker data (you can do this on a site where you've already run a migration and empty any `edd_` order related tables, then use CLI to run `wp edd migrate_payments --force` to run the order migration again anyway).

It should also work on an existing site running EDD 3.0 without going through the migration process, though, and you can test that first, before doing a migration, by viewing any order with campaign data attached to it (they should show in the payments list table). The data should be visible, but the extension/PR should silently migrate the old data to the new order metadata, and you can check that directly in the database.

**Review by LisaCee — APPROVED** [review]
- [x] 2.x migration to 3.0
- [x] 3.0 + release 1.0.2 correctly updated the meta when changed to 3.0 + this branch

**Comment by ashleyfae** [comment]
## Tested

- [x] Nothing changes in EDD 2.10
    - [x] Create campaign data in master / EDD 2.10.
    - [x] Switch to this branch.
    - [x] Ensure nothing changes.
- [ ] Data preserved during 3.0 migration
    - [ ] Create campaign data in master / EDD 2.10
    - [ ] Switch to this branch.
    - [ ] Migrate to EDD 3.0
    - [ ] Ensure data is still there.
- [ ] Data preserved when updating Campaign Tracker after 3.0 migration.
    - [ ] Create campaign data in master / EDD 2.10
    - [ ] Migrate to EDD 3.0
    - [ ] Switch to this branch.
    - [ ] Ensure data is still there.

**Review by ashleyfae — APPROVED** [review]

**Pull request merged**